src/os/bluestore/bluestore_types.cc

```diff
@@ -335,6 +335,241 @@ ostream& operator<<(ostream& out, const bluestore_extent_ref_map_t& m)
   return out;
 }
 
+// bluestore_blob_use_tracker_t
+
+void bluestore_blob_use_tracker_t::allocate()
+{
+  assert(num_au != 0);
+  bytes_per_au = new uint32_t[num_au];
+}
+
+void bluestore_blob_use_tracker_t::fall_back_to_partial(uint32_t _num_au, uint32_t _au_size)
+{
+  assert(num_au == 0 && au_size == 0);
+  assert(_num_au != 0 && _au_size != 0);
+  assert(_num_au * _au_size >= total_bytes);
```

Comment:
I think if we stay in the num_au==0 mode for small blobs (see comment below) then this assert will no longer be true